src/components/form/form.test.js

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import Form from "../form/form.component";
+import Textbox from "../textbox/textbox.component";
+import Button from "../button/button.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+import { useJQueryCssValueAndAssert } from "../../../cypress/support/component-helper/common-steps";
+import { getElement, getComponent } from "../../../cypress/locators/index";
+import {
+  formPreview,
+  formFooterComponent,
+  formButtonComponent,
+} from "../../../cypress/locators/form/index";
+
+let callback;
+beforeEach(() => {
+  callback = cy.stub();
+});
+
+const FormComponent = ({ ...props }) => {
+  return (
+    <Form
+      saveButton={
+        <Button buttonType="primary" type="submit">
+          Save
+        </Button>
+      }
+      {...props}
+    >
+      <Textbox label="Textbox1" />
+      <Textbox label="Textbox2" />
+      <Textbox label="Textbox3" />
+    </Form>
+  );
+};
+
+context("Tests for Form component", () => {
+  describe("check props for Form component", () => {
+    it.each([
+      ["left", "start"],
+      ["right", "end"],
+    ])(
+      "should check buttonAlignment set as %s for Form component",
+      (buttonAlignment, webkitAlign) => {
+        CypressMountWithProviders(
+          <FormComponent buttonAlignment={buttonAlignment} />
+        );
+        formFooterComponent().should(
+          "have.css",
+          "-webkit-box-pack",
+          webkitAlign
+        );
+      }
+    );
+
+    it("should check children for Form component", () => {
+      CypressMountWithProviders(
+        <Form>
+          <Textbox label="Textbox" />
+        </Form>
+      );
+      getElement("input").should("be.visible");
+    });
+
+    it.each([3, 11, 9])(
+      "should check errorCount as %s for Form component",
+      (errorCount) => {
+        CypressMountWithProviders(<FormComponent errorCount={errorCount} />);
+        getElement("errors")
+          .should("have.css", "color", "rgb(162, 44, 59)")
+          .and("have.text", `${errorCount} errors`)
+          .and("be.visible");
+      }
+    );
+
+    it.each([
+      [0, 287],
+      [3, 335],
+      [5, 367],
+      [7, 407],
+    ])(
+      "should check fieldSpacing as %s for Form component",
+      (fieldSpacing, formHeight) => {
+        CypressMountWithProviders(
+          <FormComponent fieldSpacing={fieldSpacing} />
+        );
+        formPreview().then(($el) => {
+          useJQueryCssValueAndAssert($el, "height", formHeight);
+        });
+      }
+    );
+
+    it("should check leftSideButtons for Form component", () => {
+      CypressMountWithProviders(
+        <FormComponent
+          leftSideButtons={
+            <Button onClick={() => console.log("cancel")} ml={2}>
```

Comment:
this is still not fixed
```suggestion
            <Button onClick={() => {}} ml={2}>
```